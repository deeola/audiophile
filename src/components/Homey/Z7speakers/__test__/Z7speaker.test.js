import ReactDOM from "react-dom";

import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import SecondSpeaker from "../ZSspeaker";

it("renders correctly in dom", () => {
  const div = document.createElement("div");

  ReactDOM.render(
    <BrowserRouter>
      <SecondSpeaker />
    </BrowserRouter>,
    div
  );
});

it("display text correctly", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <SecondSpeaker />
    </BrowserRouter>
  );

  expect(getByTestId("title")).toHaveTextContent("ZX7 SPEAKER");
});

//matches snapshot
it("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <SecondSpeaker />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
