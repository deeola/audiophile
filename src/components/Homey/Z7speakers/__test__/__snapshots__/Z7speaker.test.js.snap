// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-gsnTZi dQNMDL"
>
  <div
    className="sc-dkzDqf"
  >
    <h4
      className="sc-hKMtZM cXPvZu"
      data-testid="title"
    >
      ZX7 SPEAKER
    </h4>
    <a
      className="sc-bczRLJ cwyJdT"
      color="secondary"
      hover="secondary"
      href="/zx7"
      onClick={[Function]}
    >
      SEE PRODUCT
    </a>
  </div>
</div>
`;
